base/essentials.jl

```diff
@@ -763,6 +763,12 @@ struct Colon <: Function
 end
 const (:) = Colon()
 
+# TODO: Change lowering to do this automatically
+@eval struct Val{x}
+    (T::Type{Val{x}} where x)() = $(Expr(:new, :T))
+end
+
+
```

Comment:
@Keno This change seems unnecessary as now we merged #45062?
```suggestion
```

Reply:
Yes